index.js

```diff
@@ -328,41 +342,46 @@ module.exports = {
         // prettier-ignore
         if (err) return cb(clarify(err, `Failed to get group details when adding members`))
 
-        const content = {
-          type: 'group/add-member',
-          version: 'v2',
-          secret: secret.toString('base64'),
-          root,
-          tangles: {
-            members: {
-              root,
-              previous: [root], // TODO calculate previous for members tangle
-            },
-            // likely incorrect group tangle and this will be overwritten by
-            // publish(), we just add it here to make the spec pass
-            group: {
-              root,
-              previous: [root],
+        getRootIdOfMsgAuthor(root, (err, rootAuthorId) => {
+          // prettier-ignore
+          if (err) return cb(clarify(err, "couldn't get root id of author of root msg"))
+
+          const content = {
+            type: 'group/add-member',
+            version: 'v2',
+            groupKey: secret.toString('base64'),
+            root,
+            creator: rootAuthorId,
+            tangles: {
+              members: {
+                root,
+                previous: [root], // TODO calculate previous for members tangle
+              },
+              // likely incorrect group tangle and this will be overwritten by
+              // publish(), we just add it here to make the spec pass
+              group: {
+                root,
+                previous: [root],
+              },
             },
-          },
-          recps: [groupId, ...feedIds],
-        }
-
-        if (opts.text) content.text = opts.text
+            recps: [groupId, ...feedIds],
+          }
 
-        // TODO: this should accept bendybutt-v1 feed IDs
-        // if (!addMemberSpec(content))
-        //   return cb(new Error(addMemberSpec.errorsString))
+          if (opts.text) content.text = opts.text
 
-        findOrCreateAdditionsFeed((err, additionsFeed) => {
-          // prettier-ignore
-          if (err) return cb(clarify(err, 'Failed to find or create additions feed when adding members'))
+          if (!addMemberSpec(content))
+            return cb(new Error(addMemberSpec.errorsString))
```

Comment:
Could we move this down to between `addGroupTangle` and `publishAndPrune`, as well as remove the group tangle above, to make the checking more true to reality?

Reply:
Sounds like I a good idea